map_conversion.py

```diff
@@ -0,0 +1,206 @@
+import json
+
+import numpy as np
+from xml.dom.minidom import parseString
+import xml.etree.ElementTree as ET
+
+from argoverse.utils.manhattan_search import compute_point_cloud_bbox
+from nuscenes.map_expansion.map_api import NuScenesMap
+from nuscenes.map_expansion import arcline_path_utils
+
+
+"""
+Converts the nuScenes map into the Argoverse format.
+The nuscenes map is a json file, and is converted to an xml for use with argoverse codebase.
+
+"""
+
+filename_to_id = {'boston-seaport':'BSP_10318', 'singapore-hollandvillage':'SHV_10320',
+                  'singapore-onenorth':'SON_10322', 'singapore-queenstown':'SQT_10324'}
+def populate_lane_dict(nusc_map, root, data):
+    """
+    Return a map that links a nuscenes 'way' token to all the nodes that are a part of that way.
+    Also adds all these nodes to the xml map.
+    """
+    lane_dict = defaultdict(list)
+
+    # Map x,y value of node to the new(argo) node id in the xml
+    present_nodes = {}  # k:tuple, v: node_id
+    global_id = 0  # New id for nodes in the xml
+
+    # Loop over all lanes in the json to populate the lane to node dictionary
+    for way in data['lane']:
+        if way['token'] == '8f23d3ed-1089-4fcf-a178-a174abc69938':
+            continue
+        lane_record = nusc_map.get_lane(way['token'])  # get arcline associated with lane
+        poses = arcline_path_utils.discretize_lane(lane_record,
+                                                   resolution_meters=0.5)  # discretize the lane to given resolution
+        for pose in poses:
+            currNode = (pose[0], pose[1])
+            if currNode not in present_nodes:
+                node = ET.SubElement(root, 'node')
+                node.set('id', str(global_id))
+                present_nodes[currNode] = global_id
+                global_id += 1
+                node.set('x', str(pose[0]))
+                node.set('y', str(pose[1]))
+            lane_dict[way['token']].append(present_nodes[currNode])
+    return lane_dict
+
+def populate_polys(data):
+    """
+    Returns a dictionary poly_dict
+    poly_dict maps a lane polygon in the original map to its exterior points.
+    """
+    # Map nodes in the original json to their x, y coordinates
+    node_dict = {}  # k: node_token v: (x,y)
+
+    # Loop over all nodes in the json to populate the node_dict
+    for node in data['node']:
+        node_dict[node['token']] = (node['x'], node['y'])
+
+    # Map polygons in the original json to the x,y coordinates of their exterior nodes
+    poly_dict = {}  # k: poly_token v: np.ndarray of shape (N,2)
+
+    # Loop over all nodes in the json to populate the node_dict
+    for poly in data['polygon']:
+        poly_array = []
+        for node_id in poly['exterior_node_tokens']:
+            poly_array.append(list(node_dict[node_id]))
+        poly_dict[poly['token']] = np.array(poly_array)
+    return poly_dict
+
+def create_lanes_xml(nusc_map, root, data, filename, argo_dir, lane_dict, poly_dict):
```

Comment:
Do you mind adding type hints here, e.g. `root: str, poly_dict: Dict[str,np.ndarray]` etc?

Reply:
Done